src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs

```diff
@@ -588,59 +593,137 @@ void EmitBindCoreCall(string varName)
-                    rhs = enumValueVarName;
+                    innerExceptionTypeDisplayString = Identifier.Exception;
+                    cultureInfoTypeDisplayString = Identifier.CultureInfo;
+                    numberStylesTypeDisplayString = Identifier.NumberStyles;
                 }
-                else if (type.SpecKind == TypeSpecKind.ByteArray)
+
+                _writer.WriteBlockStart($"if ({expressionForConfigStringValue} is string {stringValueVarName})");
+
+                if (typeKind is StringParsableTypeKind.ConfigValue)
                 {
-                    rhs = $"{Expression.ConvertFromBase64String}({stringValueVarName})";
+                    EmitAssignment(expressionForMemberAccess, stringValueVarName);
+                    writeOnSuccess?.Invoke();
+                    _writer.WriteBlockEnd();
+                    return;
                 }
-                else
+                else if (typeKind is StringParsableTypeKind.Uri)
                 {
+                    string uriVarName = GetIncrementalVarName(Identifier.temp);
+                    _writer.WriteLine($"{Identifier.Uri}.{Identifier.TryCreate}({stringValueVarName}, {Identifier.UriKind}.{Identifier.RelativeOrAbsolute}, out {Identifier.Uri}? {uriVarName});");
+                    _writer.WriteBlock($$"""
+                        if ({{uriVarName}} is not null)
+                        {
+                            {{expressionForMemberAccess}} = {{uriVarName}};
+                        }
+                        """);
+                    _writer.WriteBlockEnd();
                     return;
                 }
 
-                _writer.WriteBlockStart($"if ({assignmentCondition})");
-                EmitAssignment(expressionForMemberAccess, rhs);
-                writeExtraOnSuccess?.Invoke();
+                // Types we catch exceptions for.
+                string rhs;
+                switch (typeKind)
+                {
+                    case StringParsableTypeKind.Enum:
+                        {
+                            rhs = $"({typeDisplayString}){Identifier.Enum}.{Identifier.Parse}(typeof({typeDisplayString}), {stringValueVarName}, true)";
+                        }
+                        break;
+                    case StringParsableTypeKind.ByteArray:
+                        {
+                            rhs = $"{Expression.ConvertFromBase64String}({stringValueVarName})";
+                        }
+                        break;
+                    case StringParsableTypeKind.Integer:
+                    case StringParsableTypeKind.Float:
+                        {
+                            string numberInfoKind = typeKind is StringParsableTypeKind.Integer ? Identifier.Integer : Identifier.Float;
+                            rhs = $"{typeDisplayString}.{Identifier.Parse}({stringValueVarName}, {numberStylesTypeDisplayString}.{numberInfoKind}, {cultureInfoTypeDisplayString}.{Identifier.InvariantCulture})";
```

Comment:
>{cultureInfoTypeDisplayString}.{Identifier.InvariantCulture} [](http://example.com/codeflow?start=147&length=60)

nit:
does it make sense to create a constant like `CultureInfo.InvariantCulture` instead of using 2 values?